File: PDM/PDMplugin/src/main/java/com/prolambda/pdm/PDMplugin/Dependence.java
package com.prolambda.pdm.PDMplugin;

import com.thoughtworks.xstream.annotations.XStreamAlias;

@XStreamAlias("dependence")
public class Dependence {
	
	@XStreamAlias("name")
	private String name;
	
	@XStreamAlias("version")
	private String version;
	
	@XStreamAlias("repository")
	private String repository;
	
	@XStreamAlias("platform")
	private String platform;
	
	@XStreamAlias("configuration")
	private String configuration;
	
	@XStreamAlias("location")
	private String location;
	
	
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public String getVersion() {
		return version;
	}
	public void setVersion(String version) {
		this.version = version;
	}
	public String getRepository() {
		return repository;
	}
	public void setRepository(String repository) {
		this.repository = repository;
	}
	public String getPlatform() {
		return platform;
	}
	public void setPlatform(String platform) {
		this.platform = platform;
	}
	public String getConfiguration() {
		return configuration;
	}
	public void setConfiguration(String configuration) {
		this.configuration = configuration;
	}
	public String getLocation() {
		return location;
	}
	public void setLocation(String location) {
		this.location = location;
	}
}

File: PDM/PDMBuilder/BuildTest/BuildTest/Dependence.cpp
#include "StdAfx.h"
#include "Dependence.h"


Dependence::Dependence(void)
{
}


Dependence::~Dependence(void)
{
}

string Dependence::GetServer(){
	return server;
}

void Dependence::SetServer(string server){
	this->server = server;
}

string Dependence::GetFlag(){
	return flag;
}

void Dependence::SetFlag(string flag){
	this->flag = flag;
}

string Dependence::GetRepository()
{
	return repository;
}

void Dependence::SetRepository(string repository){
	this->repository = repository;
}

string Dependence::GetName(){
	return name;
}

void Dependence::SetName(string name){
	this->name = name;
}

string Dependence::GetVersion(){
	return version;
}

void Dependence::SetVersion(string version){
	this->version = version;
}

string Dependence::GetPath(){
	return path;
}

void Dependence::SetPath(string path){
	this->path = path;
}

string Dependence::GetLocalPath(){
	return localPath;
}

void Dependence::SetLocalPath(string localPath){
	this->localPath = localPath;
}